app/controllers/application_controller.rb

```diff
@@ -1,10 +1,15 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
   before_action :configure_permitted_parameters, if: :devise_controller?
+  before_action :create_cart
 
   protected
 
   def configure_permitted_parameters
     devise_parameter_sanitizer.permit(:sign_up, keys: [:name])
   end
+
+  def create_cart
+    session[:cart_id] = ShoppingCart.create.id unless session[:cart_id]
```

Comment:
I understand the need to move on @Scharrre97, but the only reason I made this comment was to expose you guys to a different approach to do the exact same thing. 

All I expected was for you guys to at least take the time to try it out and maybe learn 1 new thing out of this. Don't forget that as much as the objective is to deliver the feature, the main thing is for you to learn something out of this experience.

This code works as it is and I have no objection to merging it.